src/components/common/data-grid/data-grid.html

```diff
@@ -0,0 +1,57 @@
+    <span class="dot placeholder"></span>
+  </ion-col>
+  <ion-col class="ion-justify-content-center display-flex" *ngFor="let header of headers">
+    <label class="text-align-center">{{header}}</label>
+  </ion-col>
+</ion-row>
+
+<br *ngIf="headers && showHeaders">
+<ion-row *ngIf="showSeparator" class="mes-row-separator"></ion-row>
+<br *ngIf="showSeparator">
+
+<ng-template *ngIf="croppedRows; then showMore else fullData;"></ng-template>
+<ng-template #showMore>
+  <ion-row *ngFor="let passedItem of croppedRows.preCrop; index as i; trackBy: trackByIndex;">
+    <ion-col [class]="accessibilityService.getTextZoomClass() === 'text-zoom-x-large' ? 'padding-top-10' : null" size="auto" *ngIf="displayColour">
+      <span [style.background-color]="finalColourArray[i]" class="dot"></span>
+    </ion-col>
+    <ion-col [class]="[y == 0 ? 'ion-justify-content-left' : 'ion-justify-content-center', 'display-flex']" *ngFor="let data of headers; index as y">
+      <span [class]="[y == 0 ? 'text-align-left' : 'text-align-center', 'mes-data']">{{passedItem[y]}}</span>
+    </ion-col>
+  </ion-row>
+
+  <ion-row
+    class="padding-top-10 display-flex ion-justify-content-center"
+    *ngIf="!showCroppedData && passedData.length > rowCropCount">
+    <ion-button class="mes-grey-button" (click)="showCroppedData = !showCroppedData">Show more</ion-button>
+  </ion-row>
+
+  <div *ngIf="showCroppedData">
+    <ion-row *ngFor="let passedItem of croppedRows.postCrop; index as i; trackBy: trackByIndex;">
+      <ion-col [class]="accessibilityService.getTextZoomClass() === 'text-zoom-x-large' ? 'padding-top-10' : null" size="auto" *ngIf="displayColour">
+        <span [style.background-color]="finalColourArray[i+croppedRows.preCrop.length]" class="dot"></span>
+      </ion-col>
+      <ion-col [class]="[y == 0 ? 'ion-justify-content-left' : 'ion-justify-content-center', 'display-flex']" *ngFor="let data of headers; index as y">
+        <span [class]="[y == 0 ? 'text-align-left' : 'text-align-center', 'mes-data']">{{passedItem[y]}}</span>
+      </ion-col>
+    </ion-row>
+  </div>
+
+  <ion-row class="padding-top-10 display-flex ion-justify-content-center" *ngIf="showCroppedData">
+    <ion-button class="mes-grey-button" (click)="showCroppedData = !showCroppedData">Show less</ion-button>
+  </ion-row>
+
+</ng-template>
+
+<ng-template #fullData>
+  <ion-row *ngFor="let passedItem of passedData; index as i; trackBy: trackByIndex;">
+    <ion-col [class]="accessibilityService.getTextZoomClass() === 'text-zoom-x-large' ? 'padding-top-10' : null" size="auto" *ngIf="displayColour">
+      <span [style.background-color]="finalColourArray[i]" class="dot"></span>
+    </ion-col>
+    <ion-col [class]="[y == 0 ? 'ion-justify-content-left' : 'ion-justify-content-center', 'display-flex']" *ngFor="let data of headers; index as y">
+      <span [class]="[y == 0 ? 'text-align-left' : 'text-align-center', 'mes-data']">{{passedItem[y]}}</span>
```

Comment:
Instead of using y == 0 to represent, we can do `let first = first;` similar to how index works